### ISSUE TITLE
lilToon Overflows System RAM on startup

### ISSUE THREAD

**Issue by Slushe** [lilxyzw/lilToon#95]
Importing lilToon **OR** having lilToon in project causes Unity to use 100% of my systems RAM until project finishes loading. Loading takes 40 to 50 minutes on average to open **WITH** lilToon. **WITHOUT** lilToon it takes 3 to 4 minutes. This causes my PC to lock up or run really slow. This happens every time I start Unity.

Project Size: 55GB

**Comment by tschaerni** [comment]
Got the same Problem while importing the 1.4.0 version.
![2023-05-27 06_45_26-Window](https://github.com/lilxyzw/lilToon/assets/5721315/49189340-0696-4c27-9fe3-057ba4b98f65)

**Comment by Slushe** [comment]
Still a problem in 1.4.1

**Comment by tschaerni** [comment]
> Still a problem in 1.4.1

I just recently installer 1.4.1 in the same project, and it works.
I don't know what the cause was. But I suppose something random like that is quite hard to debug.

**Comment by Slushe** [comment]
The only way I been able to reproduce it is by having a large project. 5-10 Booth Avatars + 10 or so outfits for each in a single project makes lilToon load every file into ram in startup or import if lilToon. I remove lilToon and it loads up just fine. I have had to resort to just forcing my projects to be small by deleting the outfit files after I finish uploading to VRChat servers which gets pretty exhausting.

**Comment by Slushe** [comment]
Unity 2022.3.6f1
LilToon 1.5.0

Still puts my 120gb Unity project into ram on startup. Happens every time I start Unity even if I let it finish the first time. 

![Unity_2023-12-10_09-07-25](https://github.com/lilxyzw/lilToon/assets/12984836/cba944b8-6e3a-4148-8b55-16c10ae30938)

**Comment by MozillaFoxy** [comment]
Unity 2019.4.31f1
Liltoon 1.5.0
Project Size: 78 GB
Same happens to me
![image3](https://github.com/lilxyzw/lilToon/assets/154413292/7896592c-6881-4000-bde6-f7576cfd503d)
![image (2)](https://github.com/lilxyzw/lilToon/assets/154413292/a5a9e7c1-7003-4db0-9a54-9e6df3966fd1)

**Comment by DarkZeyan** [comment]
Same here, since 1.4.0 I've been trying to opening my project but it crashes everytime lilToon starts to run the migrateMaterials process

**Comment by DarkZeyan** [comment]
![image](https://github.com/lilxyzw/lilToon/assets/64108165/dfc89d2f-98c8-4eca-adaf-dddcdcb842bd)
Well, by modifying this boolean to false it should fix for now the memory overflow

**Comment by lilxyzw** [comment]
My 21.3GB project doesn't have this problem, does this problem still exist?

**Comment by DarkZeyan** [comment]
> My 21.3GB project doesn't have this problem, does this problem still exist?

Yes, this problem still exists but apparently it is not directly related to the size of the project itself, I could say it is more related to the number of materials the project has and the size of the texture of those materials.

I have this problem in my 64.5 GB project and disabling Migrate on Startup has been a temporal solution for me since I modified the bool
